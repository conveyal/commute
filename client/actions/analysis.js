import moment from 'moment'
import {push} from 'react-router-redux'
import {createAction} from 'redux-actions'
import uuid from 'uuid'

const addAnalysis = createAction('add analysis')
export const createAnalysis = (newAnalysis, organizationId) => {
  newAnalysis.id = uuid.v4()
  newAnalysis.lastRunDateTime = moment().unix()
  const commuters = newAnalysis.commuters
  delete newAnalysis.commuters
  return [
    addAnalysis({ analysis: newAnalysis, organizationId }),
    push(`/organizations/${organizationId}/analysis/${newAnalysis.id}`),
    makeMockTrips(newAnalysis.id, organizationId, commuters)
  ] // TODO save to server
}

const deleteLocally = createAction('delete analysis')
/* const deleteOnServer = (id) =>
  serverAction({
    url: `/api/analysis/${id}`,
    options: {
      method: 'delete'
    }
  }) */ // TODO delete on server
export const deleteAnalysis = (id, organizationId) => [
  deleteLocally({ analysisId: id, organizationId }),
  push(`/organizations/${organizationId}`)
]

const makeMockTrips = (analysisId, organizationId, commuters) => {
  const trips = []
  const modes = ['bike', 'car', 'transit', 'walk']
  const randRange = () => Math.random() * 0.4 + 0.8
  commuters.forEach((commuter) => {
    const curTrip = {
      commuterId: commuter.id
    }
    const tripMetricBase = Math.random()
    let bestValue = 9999999
    let bestMode
    modes.forEach((mode) => {
      // generate random trips
      const modePctMultiplier = randRange()
      const cost = tripMetricBase * 34.7 * modePctMultiplier * randRange()
      const distance = tripMetricBase * 69 * modePctMultiplier * randRange()
      const time = tripMetricBase * 7500 * modePctMultiplier * randRange()
      curTrip[mode] = {cost, distance, time}
      const tripValue = cost + time * 0.01
      if (tripValue < bestValue) bestMode = mode
    })
    curTrip.mostLikely = Object.assign({ mode: bestMode }, curTrip[bestMode])
    trips.push(curTrip)
  })
  return sendMockTrips({analysisId, organizationId, trips})
}
const sendMockTrips = createAction('receive mock calculated trips')
