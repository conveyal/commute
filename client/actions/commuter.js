import {push} from 'react-router-redux'
import {createAction} from 'redux-actions'
import uuid from 'uuid'

const addCommuter = createAction('add commuter')
export const createCommuter = ({commuter, groupId, organizationId}) => {
  commuter.id = uuid.v4()
  return [
    addCommuter({commuter, groupId, organizationId}),
    push(`/organizations/${organizationId}/groups/${groupId}`)
  ] // TODO save to server
}

const deleteLocally = createAction('delete commuter')
/* const deleteOnServer = (id) =>
  serverAction({
    url: `/api/commuter/${id}`,
    options: {
      method: 'delete'
    }
  }) */ // TODO delete on server
export const deleteCommuter = ({commuterId, groupId, organizationId}) => [
  deleteLocally({commuterId, groupId, organizationId}),
  push(`/organizations/${organizationId}/groups/${groupId}`)
]

const updateLocally = createAction('update commuter')
/* const updateOnServer = (id) =>
  serverAction({
    url: `/api/commuter/${id}`,
    options: {
      method: 'update'
    }
  }) */ // TODO update on server
export const updateCommuter = ({commuter, groupId, organizationId}) => [
  updateLocally({commuter, groupId, organizationId}),
  push(`/organizations/${organizationId}/groups/${groupId}`)
]
