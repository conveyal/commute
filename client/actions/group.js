import {push} from 'react-router-redux'
import {createAction} from 'redux-actions'
import uuid from 'uuid'

const addGroup = createAction('add group')
const appendCommuters = createAction('append commuters')
export const appendToGroup = (newCommuters, groupId, organizationId) => {
  return [
    appendCommuters({ commuters: newCommuters, groupId, organizationId }),
    push(`/organizations/${organizationId}/groups/${groupId}`)
  ] // TODO save to server
}
export const createGroup = (newGroup, organizationId) => {
  newGroup.id = uuid.v4()
  newGroup.allAddressesGeocoded = false
  return [
    addGroup({ group: newGroup, organizationId }),
    push(`/organizations/${organizationId}/groups/${newGroup.id}`)
  ] // TODO save to server
}

const deleteLocally = createAction('delete group')
/* const deleteOnServer = (id) =>
  serverAction({
    url: `/api/group/${id}`,
    options: {
      method: 'delete'
    }
  }) */ // TODO delete on server
export const deleteGroup = (id, organizationId) => [
  deleteLocally({ groupId: id, organizationId }),
  push(`/organizations/${organizationId}`)
]
