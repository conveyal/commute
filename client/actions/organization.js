import {push} from 'react-router-redux'
import {createAction} from 'redux-actions'
import uuid from 'uuid'

const addLocally = createAction('add organization')
export const createOrganization = (newOrganization) => {
  newOrganization.id = uuid.v4()
  newOrganization.analyses = []
  newOrganization.analysesById = {}
  newOrganization.sites = []
  newOrganization.sitesById = {}
  newOrganization.groups = []
  newOrganization.groupsById = {}
  return [
    addLocally(newOrganization),
    push(`/organizations/${newOrganization.id}`)
  ] // TODO save to server
}

const deleteLocally = createAction('delete organization')
/* const deleteOnServer = (id) =>
  serverAction({
    url: `/api/organization/${id}`,
    options: {
      method: 'delete'
    }
  }) */ // TODO delete on server
export const deleteOrganization = (id) => [deleteLocally(id), push('/')]

export const updateOrganization = (organization) => [
  setLocally(organization),
  push(`/organizations/${organization.id}`)
]
/* const updateOnServer = (organization) =>
  serverAction({
    url: `/api/organization/${organization.id}`,
    options: {
      body: JSON.stringify(organization),
      method: 'put'
    }
  }) */  // TODO: update on server

export const setLocally = createAction('set organization')
