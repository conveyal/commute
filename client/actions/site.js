import {push} from 'react-router-redux'
import {createAction} from 'redux-actions'
import uuid from 'uuid'

const addSite = createAction('add site')
export const createSite = (newSite, organizationId) => {
  newSite.id = uuid.v4()
  return [
    addSite({ organizationId, site: newSite }),
    push(`/organizations/${organizationId}`)
  ] // TODO save to server
}

const deleteLocally = createAction('delete site')
/* const deleteOnServer = (id) =>
  serverAction({
    url: `/api/site/${id}`,
    options: {
      method: 'delete'
    }
  }) */ // TODO delete on server
export const deleteSite = (id, organizationId) => [
  deleteLocally({ siteId: id, organizationId }),
  push(`/organizations/${organizationId}`)
]

const updateLocally = createAction('update site')
/* const updateOnServer = (id) =>
  serverAction({
    url: `/api/site/${id}`,
    options: {
      method: 'update'
    }
  }) */ // TODO update on server
export const updateSite = (site, organizationId) => [
  updateLocally({ organizationId, site }),
  push(`/organizations/${organizationId}`)
]
