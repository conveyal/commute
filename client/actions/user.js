import {createAction} from 'redux-actions'

// login / logout
export const login = createAction('log in')
export const logout = createAction('log out')
