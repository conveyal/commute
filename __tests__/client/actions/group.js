/* global describe, it */

import {expectCreateGroup, expectDeleteGroup} from '../../test-utils/actions'

import * as group from '../../../client/actions/group'

describe('actions > group', () => {
  it('create group should work', () => {
    const data = {
      name: 'Mock Group'
    }
    const result = group.createGroup(data, '1')

    expectCreateGroup(result)
  })

  it('delete group should work', () => {
    const result = group.deleteGroup('1', '1')

    expectDeleteGroup(result)
  })
})
