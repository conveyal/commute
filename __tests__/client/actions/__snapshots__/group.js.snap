exports[`actions > group create group should work 1`] = `
Object {
  "payload": Object {
    "group": Object {
      "allAddressesGeocoded": false,
      "name": "Mock Group",
    },
    "organizationId": "1",
  },
  "type": "add group",
}
`;

exports[`actions > group create group should work 2`] = `
Object {
  "payload": Object {
    "args": Array [
      "/organizations/1/groups/new-entity-id",
    ],
    "method": "push",
  },
  "type": "@@router/CALL_HISTORY_METHOD",
}
`;

exports[`actions > group delete group should work 1`] = `
Array [
  Object {
    "payload": Object {
      "groupId": "1",
      "organizationId": "1",
    },
    "type": "delete group",
  },
  Object {
    "payload": Object {
      "args": Array [
        "/organizations/1",
      ],
      "method": "push",
    },
    "type": "@@router/CALL_HISTORY_METHOD",
  },
]
`;
